### PR TITLE
Added Deployment Guide 2.0

### PR DESCRIPTION
**Pull request by TusharPatel44**
# Description

## Type of change

- [ ] *Bug fix (non-breaking change which fixes an issue)*
- [ ] *New feature (non-breaking change which adds functionality)*
- [ ] *Breaking change (fix or feature that would cause existing functionality to not work as expected)*
- [ ] *Refactor* (existing code being refactored)
- [ ] *This change requires a documentation update*

# Submission Checklist

- [ ] Build Verified
- [ ] Bundle Verified
- [ ] Lint passing
- [ ] Unit tests passing
- [ ] Integration tests passing
- [ ] Change logs generated

### CODE REVIEW

File: source/docs/deployment-guide.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        Rush bundle
+ 
+2.	Run Deployment command. This command will deploy necessary CDF stacks  to AWS cloud formation.
+
+	a. Following command deploys asset library lite version + services that has configuration defined:
+        
+            ./infrastructure/deploy-core.bash -e ENV_NAME \
+            -b BUCKET-NAME \
+            -p EC2_KEYPAIR_NAME \
```
Comment (jonasneu-aws) [inline source/docs/deployment-guide.md:113]: The `-p` argument for `deploy-core.bash` is vestigal and no longer gets used in recent versions of CDF.

**Review by jonasneu-aws — CHANGES_REQUESTED** [review]
Hi @TusharPatel44, thank you for contributing to the CDF docs! 

This PR does introduce some duplication with [deployment.md](https://github.com/aws/aws-connected-device-framework/blob/main/source/docs/deployment.md) and [prerequisites.md](https://github.com/aws/aws-connected-device-framework/blob/main/source/docs/development/prerequisites.md). Would you mind merging the new information added here with the existing materials?

File: source/packages/services/assetlibrary/infrastructure/cfn-assetLibrary.yaml [hunk 2]
```diff
@@ -220,6 +220,7 @@ Resources:
           - Action:
               - "iot:DescribeThingType"
               - "iot:CreateThingType"
+              - "iot:DeprecateThingType"
```
[the remark this answers is not in the document]
Comment (TusharPatel44) [reply source/packages/services/assetlibrary/infrastructure/cfn-assetLibrary.yaml:223]: Yes This is bug fix. It is missing required permission to perform deprecate thing operation.

File: source/docs/deployment-guide.md (hunk 1 above)
Comment (TusharPatel44) [reply source/docs/deployment-guide.md:113]: Got it. I will update guide.

**Comment by TusharPatel44** [comment]
> Hi @TusharPatel44, thank you for contributing to the CDF docs!
> 
> This PR does introduce some duplication with [deployment.md](https://github.com/aws/aws-connected-device-framework/blob/main/source/docs/deployment.md) and [prerequisites.md](https://github.com/aws/aws-connected-device-framework/blob/main/source/docs/development/prerequisites.md). Would you mind merging the new information added here with the existing materials?

I will make changes and create new PR.

**Pull request closed without merging**